edifice/_component.py

```diff
@@ -606,11 +651,33 @@ def __init__(self, *args: P.args, **kwargs: P.kwargs):
             super().__init__()
 
         def render(self):
-            # We cannot type this because PropsDict forgets the types
-            return f(self, **self.props._d) # type: ignore[reportGeneralTypeIssues]
+            with Container() as root:
```

Comment:
What is this recursively defined `Container`?

Reply:
A `Container` is a simple `Component` that renders to nothing. It's used to track the `children` so that we have a context manager that can keep track of the elements rendered to a functional component.